db/transaction.go

```diff
@@ -0,0 +1,157 @@
+package db
+
+import (
+	"bytes"
+	"errors"
+	"fmt"
+
+	"github.com/attic-labs/noms/go/types"
+
+	"roci.dev/diff-server/kv"
+	nomsjson "roci.dev/diff-server/util/noms/json"
+	"roci.dev/diff-server/util/time"
+)
+
+var (
+	// ErrClosed is the error returned from operations on a Transaction when
+	// it has already been closed.
+	ErrClosed = errors.New("Transaction is closed")
+)
+
+type Transaction struct {
+	db     *DB
+	head   Commit
+	me     *kv.MapEditor
+	wrote  bool
+	closed bool
+	name   string
+	args   types.List
+}
+
+// Closed returns true when the database has been closed. A database becomes
```

Comment:
possible search and replace misfire? i think you mean transaction has been close and a transaction becomes closed, not a database...?